packages/nodes-base/nodes/Code/Sandbox.ts

```diff
@@ -19,11 +19,16 @@ export interface SandboxContext extends IWorkflowDataProxyData {
 export const REQUIRED_N8N_ITEM_KEYS = new Set(['json', 'binary', 'pairedItem', 'error']);
 
 export function getSandboxContext(this: IExecuteFunctions, index: number): SandboxContext {
+	const helpers = {
+		...this.helpers,
+		httpRequestWithAuthentication: this.helpers.httpRequestWithAuthentication.bind(this),
+		requestWithAuthenticationPaginated: this.helpers.requestWithAuthenticationPaginated.bind(this),
```

Comment:
Works great for these helpers, should we do it for all?

```js
const helpers = mapValues(this.helpers, (helper) => helper.bind(this));
```

Reply:
Wouldn't requestWIthAuthentication still fail as I think it uses the passed in node parameters to work out which credential type to use which the code node doesn't have access to?

Reply:
@elsmr I explicitly didn't want to bind any other helper because most helper methods don't need a bound `this`, and the others that do are deprecated.
 
@Joffcom `this.helpers.requestWIthAuthentication` does not take those additional parameters. They are picked from a bound outer scope. https://github.com/n8n-io/n8n/blob/master/packages/core/src/NodeExecuteFunctions.ts#L3129